libcudacxx/include/cuda/memory_resource

```diff
@@ -80,559 +80,22 @@ class resource_ref {
 */
 // clang-format on
 
-#ifdef LIBCUDACXX_ENABLE_EXPERIMENTAL_MEMORY_RESOURCE
```

Comment:
Suggestion: I really think the refactoring in this PR should be in a separate PR from the new features added (cuda_memory_resource).